src/BLL/Managers/TopicManager.cs

```diff
@@ -0,0 +1,195 @@
+                        dbContext.Entry(topic).State = EntityState.Modified;
+                        
+                        // Deassociated previous Links
+                        DeassociateAllLinksFromTopic(topicId);
+                        await dbContext.SaveChangesAsync();
+
+                        // Add all associations
+                        AssociateUsersToTopicByRole(topic.Id, Role.Student, model.Students);
+                        AssociateUsersToTopicByRole(topic.Id, Role.Supervisor, model.Supervisors);
+                        AssociateUsersToTopicByRole(topic.Id, Role.Reviewer, model.Reviewers);
+                        AssociateTopicsToTopic(topic.Id, model.AssociatedTopics);
+
+                        await dbContext.SaveChangesAsync();
+
+                        transaction.Commit();
+                        return true;
+                    }
+                    catch(Exception)
+                    {
+                        transaction.Rollback();
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        public virtual async Task<bool> DeleteTopicAsync(int topicId)
+        {
+            var topic = await dbContext.Topics.AsNoTracking().FirstOrDefaultAsync(u => u.Id == topicId);
+            
+            if (topic != null)
+            {
+                DeassociateAllLinksFromTopic(topicId);
+                dbContext.Remove(topic);
+                await dbContext.SaveChangesAsync();
+
+                return true;
+            }
+
+            return false;
+        }
+
+    
+        // Private Region
+
+        private void AssociateUsersToTopicByRole(int topicId, string role, int[] userIds)
+        {
+            if (userIds != null)
+            {
+                foreach (int userId in userIds)
+                {
+                    dbContext.TopicUsers.Add(new TopicUser() 
+                    { 
+                        UserId = userId, 
+                        TopicId = topicId, 
+                        Role = role 
+                    });
+                }
+            }
+        }
+
+        private void AssociateTopicsToTopic(int topicId, int[] associatedTopicIds)
+        {
+            if (associatedTopicIds != null)
+            {
+                foreach (int associatedTopicId in associatedTopicIds)
+                {
+                    dbContext.AssociatedTopics.Add(new AssociatedTopic() 
+                    { 
+                        ChildTopicId = topicId, 
+                        ParentTopicId = associatedTopicId
+                    });
+                }
+            }
+        }
+
+        private void DeassociateAllLinksFromTopic(int topicId)
+        {
+            dbContext.TopicUsers.RemoveRange(dbContext.TopicUsers.Where(tu => tu.TopicId == topicId));
+            dbContext.AssociatedTopics.RemoveRange(dbContext.AssociatedTopics.Where(tu => tu.ParentTopicId == topicId));
+        }
+    }
+}
```

Comment:
Like, in case of delete I would just delete the record in one of the table then the associated record in the intermediate table automatically deletes(cascades). 
And while adding I don't need to check for referential integrity. That's what I meant from my question